### PR TITLE
CI fix: pin url for msrv build

### PR DESCRIPTION
**Pull request by lalitb**
Attempt to fix the CI build issue arising from new published `url` version
https://github.com/servo/rust-url/issues/992

```
    Checking idna v1.0.3
error[E0658]: use of unstable library feature 'error_in_core'
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/idna-1.0.3/src/lib.rs:78:6
   |
78 | impl core::error::Error for Errors {}
   |      ^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #103765 <https://github.com/rust-lang/rust/issues/103765> for more information

For more information about this error, try `rustc --explain E0658`.
```

### CODE REVIEW

**Comment by lalitb** [comment]
Oh, it's not just the msrv failure, the CI will fail for opentelemetry-sdk too for rustc 1.70.0 and `jaeger_remote_sampler` feature :
```

labhas@DESKTOP-KNU1IS8:~/obs/ot/lalit/rust/opentelemetry-rust/opentelemetry-sdk$ cargo build --features="jaeger_remote_sampler"
   Compiling smallvec v1.13.2
   Compiling serde v1.0.214
   Compiling form_urlencoded v1.2.1
   Compiling futures-util v0.3.31
   Compiling rand_chacha v0.3.1
   Compiling opentelemetry-http v0.26.0 (/home/labhas/obs/ot/lalit/rust/opentelemetry-rust/opentelemetry-http)
   Compiling icu_normalizer v1.5.0
   Compiling rand v0.8.5
   Compiling idna_adapter v1.2.0
   Compiling idna v1.0.3
error[E0658]: use of unstable library feature 'error_in_core'
  --> /home/labhas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/idna-1.0.3/src/lib.rs:78:6
   |
78 | impl core::error::Error for Errors {}
   |      ^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #103765 <https://github.com/rust-lang/rust/issues/103765> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `idna` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...

```

better to pin the Cargo.toml

**Comment by lalitb** [comment]
> better to pin the Cargo.toml

File: Cargo.toml [hunk 1]
```diff
@@ -47,4 +47,4 @@ tokio-stream = "0.1"
 tracing = { version = "0.1", default-features = false }
 tracing-core = { version = "0.1", default-features = false }
 tracing-subscriber = { version = "0.3", default-features = false }
-url = { version = "2.5", default-features = false }
+url = { version = "2.5.2", default-features = false } #https://github.com/servo/rust-url/issues/992
```
Comment (cijothomas) [inline Cargo.toml:50]: should we do a changelog entry for those crates affected?

Comment (lalitb) [reply Cargo.toml:50]: yes we should. Have updated the otel-sdk changelog.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**